presto-main/src/main/java/com/facebook/presto/type/ArrayType.java

```diff
@@ -65,7 +65,30 @@ public boolean isOrderable()
     @Override
     public boolean equalTo(Block leftBlock, int leftPosition, Block rightBlock, int rightPosition)
     {
-        return compareTo(leftBlock, leftPosition, rightBlock, rightPosition) == 0;
+        if (isOrderable()) {
+            return compareTo(leftBlock, leftPosition, rightBlock, rightPosition) == 0;
+        }
```

Comment:
You don't need this `if`. It doesn't quite make sense for orderable non-comparable type to exist.